packages/analytics-browser/src/browser-client.ts

```diff
@@ -190,21 +189,32 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
 
     if (isSessionTrackingEnabled(this.config.defaultTracking)) {
       if (previousSessionId && lastEventTime) {
-        this.track(DEFAULT_SESSION_END_EVENT, undefined, {
+        await this.track(DEFAULT_SESSION_END_EVENT, undefined, {
```

Comment:
Instead of waiting on this result here, I think it would be better to return this promise so users can choose to await or ignore.

Reply:
See [comment](https://github.com/amplitude/Amplitude-TypeScript/pull/696#discussion_r1560203134).

Reply:
Removed the await and make setSessionId return a promise.